### DISCUSSION TITLE
Typesafe and expected result.

### DISCUSSION THREAD

**Discussion by H7ioo** [SBoudrias/Inquirer.js#1295] [Q&A]
Is there a way to type typesafe prompts? For example: 
```js
  const res = await prompt<{ title: string }>([
    { type: "input", name: "titlee" },
  ]);
```
type of res is 
```js
const res: {
    title: string;
}
```
but actually, it's not because I misspelled the name key. Also, if I want to filter the input.
```js
  const res = await prompt<{ title: string }>([
    {
      type: "input",
      name: "title",
      filter: (input: string) => {
        return parseInt(input);
      },
    },
  ]);
```
The input is a number but the typescript tells me that it's a string. So is there a way to enforce return type to filter or have typesafe names?

**Comment by SBoudrias** [comment]
Hey @H7ioo, these types are from `@types/inquirer` which aren't supported by this team ([they're here](https://github.com/DefinitelyTyped/DefinitelyTyped)).

I'd highly encourage you to migrate to the new Inquirer package which is written in Typescript and officially supported. See #1214 for the new version notes; or refer to the root README.

**Reply by H7ioo** [reply]
Thanks, I checked on the new library, does it support an array of prompts or is it just 1 prompt at a time. I didn't find an example on the array of prompts so I'm assuming it has a different approach.

**Reply by SBoudrias** [reply]
@H7ioo yes, it only supports 1 prompt at a time. But check the example recipes if you want an answer object: https://github.com/SBoudrias/Inquirer.js#get-answers-in-an-object

**Reply by H7ioo** [reply]
I appreciate your help. I'll definitely check it out.
